planner/core/rule_column_pruning.go

```diff
@@ -149,7 +149,21 @@ func (la *LogicalAggregation) PruneColumns(parentUsedCols []*expression.Column)
 			la.GroupByItems = []expression.Expression{expression.NewOne()}
 		}
 	}
-	return child.PruneColumns(selfUsedCols)
+	err := child.PruneColumns(selfUsedCols)
+	if err != nil {
+		return err
+	}
+	// Do an extra Projection Elimination here. This is specially for empty Projection below Aggregation.
+	// This kind of Projection would cause some bugs for MPP plan and is safe to be removed.
+	// This kind of Projection should be removed in Projection Elimination, but currently PrunColumnsAgain is
+	// the last rule. So we specially handle this case here.
+	if childProjection, isProjection := child.(*LogicalProjection); isProjection {
+		if len(childProjection.Exprs) == 0 && childProjection.Schema().Len() == 0 {
+			childOfChild := childProjection.children[0]
+			la.SetChildren(childOfChild)
+		}
+	}
+	return nil
```

Comment:
why not put these into the postOptimize() which eliminates projection at last?

Reply:
```
func postOptimize(sctx sessionctx.Context, plan PhysicalPlan) PhysicalPlan {
	plan = eliminatePhysicalProjection(plan)
...
```

Reply:
According to https://github.com/pingcap/tics/issues/3389#issuecomment-963852137, this `Projection` is indeed eliminated in the `postOptimize()`. But the issue is that it affects the physical optimization and we have to eliminate it before physical optimization.